### PR TITLE
Port to amp v3

### PR DESCRIPTION
**Pull request by danog**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,23 +23,25 @@
...
-        "amphp/amp": "^2.0.3",
-        "amphp/parallel": "^1.4",
-        "amphp/serialization": "^1.0",
-        "laravel/serializable-closure": "^1.0"
+        "php": ">=8.1",
+        "amphp/amp": "^3",
+        "amphp/parallel": "^2",
+        "amphp/serialization": "^1",
+        "laravel/serializable-closure": "^1.0",
+        "psalm/phar": "^5.13"
```
Comment (kelunik) [inline composer.json]: Psalm should be on require-dev.

Comment (danog) [reply composer.json]: Whoops :)

File: test/MapTest.php [hunk 2]
```diff
@@ -1,37 +1,29 @@
...
-
-    public function testCorrectOutputOrder()
-    {
-        $this->assertSame([0, 1, 0], wait(parallelMap([0, 1, 0], function ($input) {
-            \sleep($input);
-
-            return $input;
-        })));
+        });
+        ksort($r);
```
Comment (kelunik) [inline test/MapTest.php]: @trowski Too bad we no longer keep the order.

File: test/ParallelTest.php [hunk 3]
```diff
@@ -93,46 +86,46 @@ public function __invoke()
...
 
     public function testUnserializableClassStaticMethod()
     {
         $this->expectException(\Error::class);
         $this->expectExceptionMessage(
             PHP_VERSION_ID >= 80000 ?
-                'Uncaught Error in worker with message "Class "Amp\\ParallelFunctions\\Test\\UnserializableClass" not found"' :
-                'Uncaught Error in worker with message "Class \'Amp\\ParallelFunctions\\Test\\UnserializableClass\' not found"'
+                'Error thrown in context with message "Class "Amp\\ParallelFunctions\\Test\\UnserializableClass" not found" and code "0"' :
+                'Error thrown in context with message "Class \'Amp\\ParallelFunctions\\Test\\UnserializableClass\' not found" and code "0"'
```
Comment (kelunik) [inline test/ParallelTest.php]: We can drop the < PHP 8 branch here.

File: test/MapTest.php (hunk 2 above)
Comment (trowski) [reply test/MapTest.php]: I guess we could sort again based on the original array.

Comment (danog) [reply test/MapTest.php]: Shall I implement this?

Comment (danog) [reply test/MapTest.php]: Done!

**Review by kelunik — APPROVED** [review]

**Review by trowski — APPROVED** [review]

**Pull request merged**